npm/webpack-dev-server/src/helpers/angularHandler.ts

```diff
@@ -28,14 +30,53 @@ type AngularJson = {
   }
 }
 
+export type ProjectConfig = {
+  root: string
+  sourceRoot: string
+  buildOptions: BuildOptions
```

Comment:
If a user is explicitly supplying `buildOptions` we don't need to take into account the `configurations` since we will still have the problem of assuming `configurations.development` which might not exist. I think this option should only contain the browser builder options and we let the user determine what configuration they'd like to merge options in, something like:
```js
const { projects } = require('./angular.json')

const projectConfig = {
  root: projects.custom.root,
  sourceRoot: projects.custom.sourceRoot,
  buildOptions: {
    ...projects.custom.architect.build.options
    ...projects.custom.architect.build.configurations.test
  }
}

module.exports = { 
  component: {
    devServer: {
      framework: 'angular',
      bundler: 'webpack',
      options: {
        projectConfig
      }
    }
  }
}
```

Since this is an escape hatch, I like the idea of no longer relying on convention (like the expected shape of a proejct config target)